app/controllers/recherche_controller.rb

```diff
@@ -28,6 +28,15 @@ def index
     @dossiers_count = matching_dossiers_ids.count
     @followed_dossiers_id = current_instructeur&.followed_dossiers&.where(id: @paginated_ids)&.ids || []
     @dossier_avis_ids_h = current_expert&.avis&.where(dossier_id: @paginated_ids)&.pluck(:dossier_id, :id).to_h || {}
+
+    # we want to retrieve dossiers that are not accessible to the instructor because he is not in the instructor group
+    # to display an alert in the view
+    instructeur_procedure_dossiers_ids = DossierSearchService
+      .matching_dossiers(current_instructeur&.procedures&.map(&:dossiers)&.first, @search_terms, with_annotation: true)
```

Comment:
je suis etonné de voir un (&:dossiers)&.first [en gros on prend les dossiers de la 1ere procédure]
plutot qqch comme ?


```suggestion
      .matching_dossiers(Dossier.joins(:procedure).where(procedure: {id: current_instructeur.procedures.ids}), @search_terms, with_annotation: true)
```

Reply:
Ah oui ça semble en effet plus clair dans ce sens.
Avec le `.first` mon idée était de "rentrer" dans le 1er niveau de tableau car j’atterrissais avec un tableau de tableau - mais je me suis peut-être trompée du coup.

Reply:
`.flatten` pour applatir un tableau de tableau. 
ex: `[[:a], [:b], [:c]].flatten` => `[:a, :b, :c]`. 
dans le cas du `.first`, `[[liste des dossiers procedure 1], [liste des dossiers procedure 2], [liste des dossiers procedure 3]].first` => `[liste des dossiers procedure 1]`

Reply:
Merci pour les explications, j'avais bien en-tête d'utiliser `.flatten`pour ce genre de cas mais ça me renvoyait une erreur : `undefined method visible_by_administration' for [#<Dossier id: 2, api_entreprise_job_exceptions: nil, archived: false, autorisation_donnees: true, brouillon_close_to_expiration_notice_sent_at: nil, conservation…`
D'où mon usage, un peu désespéré du `.first` :)

Reply:
@mfo si je me gourre pas, on file une relation a DossierSearchService qui est après affiné, je pense pas qu'on ait un gain avec une liste d'id.

je me demande s'il ne faudrait pas faire une première recherche sur tous les dossiers des procédures ... puis réduire.

```
dossier_for_all_groups = Dossier.where(groupe_instructeur: GroupeInstructeur.where(procedure: sim.instructeur.procedures))

visible_dossiers, hidden_dossier = dossier_for_all_groups.partition { |d| current_instructeur.groupe_instructeurs.pluck(:id).include?(d.groupe_instructeur_id) }
```
ou un truc comme ca ?

Alors apres vient le probleme de la perf effectivement, par ce que la on part sur une recherche sur tous les dossiers ...

Sinon on peut essayer de chercher sur tout les groupes uniquement si le `search_term` est un id

Reply:
@LeSim on a simplifié un peu la feature en faisant les requetes que si la recherche est faite sur un id. Je te laisse me dire ce que tu en penses.